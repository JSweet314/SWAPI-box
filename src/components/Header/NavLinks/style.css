.nav {
  padding: 1rem 0;
  display: flex;
  justify-content: space-around;
  width: 100%;
}

.nav li {
  display: inline;
}

.nav a { 
  text-decoration: none;
  background-color: gray;
  border-radius: 10px 10px;
  padding: 1rem 2rem;
  color: #fff;
}

.nav .active {
  background-color: #18212A;
}