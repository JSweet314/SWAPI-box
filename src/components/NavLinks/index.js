import React from 'react';
import {NavLink} from 'react-router-dom';
import './style.css';

const NavLinks = () => {
  return (
    <ul className="nav">
      <li>
        <NavLink exact to="/">Home</NavLink>
      </li>
      <li>
        <NavLink exact to="/people">People</NavLink>
      </li>
      <li>
        <NavLink exact to="/planets">Planets</NavLink>
      </li>
      <li>
        <NavLink exact to="/vehicles">Vehicles</NavLink>
      </li>
    </ul>
  );
};

export default NavLinks;