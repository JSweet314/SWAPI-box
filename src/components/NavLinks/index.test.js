import React from 'react';
import {shallow} from 'enzyme';
import NavLinks from './index';

it('should match a snapshot', () => {
  const wrapper = shallow(<NavLinks />);
  expect(wrapper).toMatchSnapshot();
});