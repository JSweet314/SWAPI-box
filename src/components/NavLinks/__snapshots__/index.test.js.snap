// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavLinks />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    exact={true}
                    to="/"
          >
                    Home
          </NavLink>
</li>,
        <li>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    exact={true}
                    to="/people"
          >
                    People
          </NavLink>
</li>,
        <li>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    exact={true}
                    to="/planets"
          >
                    Planets
          </NavLink>
</li>,
        <li>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    exact={true}
                    to="/vehicles"
          >
                    Vehicles
          </NavLink>
</li>,
      ],
      "className": "nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            ariaCurrent="true"
            exact={true}
            to="/"
>
            Home
</NavLink>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Home",
            "exact": true,
            "to": "/",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            ariaCurrent="true"
            exact={true}
            to="/people"
>
            People
</NavLink>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "People",
            "exact": true,
            "to": "/people",
          },
          "ref": null,
          "rendered": "People",
          "type": [Function],
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            ariaCurrent="true"
            exact={true}
            to="/planets"
>
            Planets
</NavLink>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Planets",
            "exact": true,
            "to": "/planets",
          },
          "ref": null,
          "rendered": "Planets",
          "type": [Function],
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            ariaCurrent="true"
            exact={true}
            to="/vehicles"
>
            Vehicles
</NavLink>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Vehicles",
            "exact": true,
            "to": "/vehicles",
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": [Function],
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        exact={true}
                        to="/"
            >
                        Home
            </NavLink>
</li>,
          <li>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        exact={true}
                        to="/people"
            >
                        People
            </NavLink>
</li>,
          <li>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        exact={true}
                        to="/planets"
            >
                        Planets
            </NavLink>
</li>,
          <li>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        exact={true}
                        to="/vehicles"
            >
                        Vehicles
            </NavLink>
</li>,
        ],
        "className": "nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              exact={true}
              to="/"
>
              Home
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Home",
              "exact": true,
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              exact={true}
              to="/people"
>
              People
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "People",
              "exact": true,
              "to": "/people",
            },
            "ref": null,
            "rendered": "People",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              exact={true}
              to="/planets"
>
              Planets
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Planets",
              "exact": true,
              "to": "/planets",
            },
            "ref": null,
            "rendered": "Planets",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              exact={true}
              to="/vehicles"
>
              Vehicles
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Vehicles",
              "exact": true,
              "to": "/vehicles",
            },
            "ref": null,
            "rendered": "Vehicles",
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
